safe-global/safe-wallet-web
# fix: Cleanup pending txs if nonce already executed

## What it solves

Follow up to #4025

This is needed in order for #4069 to work since migration transactions are always replaced and their pending state is stuck indefinitely which bricks the safe.

## How this PR fixes it

- Stores the `nonce` for each pending transaction
- Extends the pending tx cleanup logic in the tx history middleware by not just checking if there is a matchin txId but also if there is a matching nonce and in case there is and txIds don't match it means the pending transaction got replaced so it is safe to clear it

## ToDo

- [x] Extend tests for `txHistorySlice`
- [x] (optional) Simplify test mocks by using a builder

## How to test it

1. Open the same safe in two different browsers
2. Execute a transaction with the same nonce at the same time from each browser
3. Observe both pending states to be cleared as soon as one of the transactions is executed

## Screenshots

## Checklist
